packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -18,6 +18,7 @@ import { traceFunction } from "./support";
 export type TransactionRun<T> = (transaction: GadgetTransaction) => Promise<T>;
 export interface GadgetSubscriptionClientOptions extends Partial<SubscriptionClientOptions> {
   urlParams?: Record<string, string | null | undefined>;
+  connectionTimeout?: number;
```

Comment:
I think we should suffix with `Ms`, like you did with the timeout arg below, since this is "external" facing (in other words, far more important to be clear about unit here).